ripley/api/util.py

```diff
@@ -97,12 +97,17 @@ def csv_download_response(rows, filename, fieldnames=None):
 
 
 def start_login_session(user):
+    user_str = f'UID {user.uid}, canvas_user_id {user.canvas_user_id}, canvas_site_id {user.canvas_site_id}'
+    current_user_str = f'UID {current_user.uid}, Canvas ID {current_user.canvas_user_id}, course ID {current_user.canvas_site_id}'
+    if current_user.is_authenticated and current_user.uid == user.uid:
+        app.logger.debug(f'User ({user_str}) has existing Ripley session: {current_user_str}')
+    else:
+        if current_user.is_authenticated:
+            app.logger.info(f"""User ({user_str}) does not match existing session ({current_user_str}). Terminating existing session.""")
+            logout_user()
+        else:
+            app.logger.debug(f'User ({user_str}) has no existing Ripley session')
```

Comment:
@lyttam What will this logic do if there's an existing session where the UID matches but the canvas site id doesn't? Do we still need to log out in that event, or will the login call below take care of it?

Reply:
@pauline2k I think it will result in 2 sessions with the same UID, but different site IDs - which is what I was going for, but I just noticed `User._get_cache_key` has only the UID. That could be the root of this whole problem because of the way the LTI launch utilizes cache storage to get around the cookie issue.